core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -291,8 +292,12 @@ object SparkSubmit {
       } else {
         Nil
       }
+
+    val ivySettings = Option(args.ivySettingsFile).map(SparkSubmitUtils.loadIvySettings).getOrElse(
```

Comment:
I just figured any repositories would be added to the settings file, but I can see how having both options could be of some benefit so let me change that.